### PR TITLE
[Feature] Create `WorkStream` model

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves partially, #7574 

## 👋 Introduction

This is setup for the new `WorkStream` migration.

## 🕵️ Details

This is a partial completion of the migration to the new `WorkStream` model. It does the following:

- Migrations with relationships in the tables
- New `WorkStream` model
- Policy for new model along with permissions
- Seeds current `PoolStream` enum values into `work_streams` table
- New relationships, leaving existing ones as is
- Queries and mutations to the graphql schema
- Tests for the new mutations and queries

## 🧪 Testing

1. Confirm the database migrations including relationships match the expectations outlined in the ticket
2. Confirm the data seeds properly
3. Confirm permissions meet the requirements
4. Confirm tests have decent coverage and consistently pass

### CODE REVIEW

File: api/database/seeders/DatabaseSeeder.php [hunk 1]
```diff
@@ -15,13 +15,13 @@ public function run(): void
 
         $this->call([
             // standard platform data
-            RolePermissionSeeder::class,
-            ClassificationSeeder::class,
+            RolePermissionSeeder::class, ClassificationSeeder::class,
```
Comment (vd1992) [inline api/database/seeders/DatabaseSeeder.php:18]: Accidental?

File: api/database/migrations/2024_11_29_163716_work_stream_relationships.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    /**
+     * Reverse the migrations.
+     */
+    public function down(): void
+    {
+        Schema::table('pools', function (Blueprint $table) {
+            $table->dropForeign('work_stream_id');
+        });
```
Comment (vd1992) [inline api/database/migrations/2024_11_29_163716_work_stream_relationships.php:46]: Reversing migration appears to fails for me here

Comment (esizer) [reply api/database/migrations/2024_11_29_163716_work_stream_relationships.php]: Good catch, fixed in [fix migration rollback](https://github.com/GCTC-NTGC/gc-digital-talent/pull/12152/commits/31cd364b91da7b6d064b91304da98416b72e7d87)

**Comment by tristan-orourke** [comment]
Querying for WorkStreams caused an error, due to lazy loading Community

image

**Review by tristan-orourke — CHANGES_REQUESTED** [review]
Everything looks good, except the lazy loading error

File: api/database/factories/JobPosterTemplateFactory.php [hunk 3]
```diff
@@ -36,6 +37,11 @@ public function definition(): array
             $classification = Classification::factory()->create();
         }
 
+        $workStream = WorkStream::inRandomOrder()->first();
+        if (! $workStream) {
+            $workStream = WorkStream::factory()->create();
+        }
```
[the remark this answers is not in the document]
Comment (esizer) [reply api/database/factories/JobPosterTemplateFactory.php]: Not guaranteed to match, but in most cases it should now after [attempt to match pool and work stream](https://github.com/GCTC-NTGC/gc-digital-talent/pull/12152/commits/d39ab2cdc446f488613c34af006934b7b8b675e6)

File: api/database/factories/PoolFactory.php [hunk 4]
```diff
@@ -187,6 +188,13 @@ public function draft(): Factory
             // the base state is draft already
             $hasSpecialNote = $this->faker->boolean();
             $isRemote = $this->faker->boolean();
+            $workStreamId = WorkStream::inRandomOrder()
+                ->limit(1)
+                ->pluck('id')
+                ->first();
```
[the remark this answers is not in the document]
Comment (esizer) [reply api/database/factories/PoolFactory.php]: I think I just copy and pasted from somewhere else :sweat_smile: Updated to be similar (and simpler) in [simplify work stream query](https://github.com/GCTC-NTGC/gc-digital-talent/pull/12152/commits/0f59ad93681923784106f2d9074e15ab0c7a5f23)

**Comment by esizer** [comment]
> Querying for WorkStreams caused an error, due to lazy loading Community

Oops, I totally forgot to add the relationship in the schem :stuck_out_tongue: Fixed in [eager load community relationship](https://github.com/GCTC-NTGC/gc-digital-talent/pull/12152/commits/0acfaa9affdb77f347ae6bf7dcfa22367b41420b)

**Review by tristan-orourke — APPROVED** [review]

**Pull request merged**